eventhandlers/action_runners/actions/VertexTrainDatasetAction.py

```diff
@@ -0,0 +1,291 @@
+                dataset_file_list.append({
+                    "filename": f"gs://{self.action.config_data.get('staging_bucket_name_without_gs_prefix')}/{file_link}",
+                    "diffgram_file": file
+                })
+
+        self.count_dataset_labels(session, dataset_file_list)
+        vertexai_import_file = self.write_vertex_format_jsonl_file(gcp_data_tools, temp_folder_name, dataset_file_list)
+
+        self.clean_up_temp_dir(temp_folder_name)
+
+        datasets_list = aiplatform.datasets.ImageDataset.list()
+        existing_datasets = []
+
+        for dataset in datasets_list:
+            existing_datasets.append(dataset.__dict__['_gca_resource'].__dict__['_pb'].display_name)
+
+        working_dataset = None
+        if self.action.config_data.get('model_name') in existing_datasets:
+            dataset_index = existing_datasets.index(self.action.config_data.get('model_name'))
+            working_dataset = datasets_list[dataset_index]
+
+            working_dataset.delete()
+
+        working_dataset = aiplatform.ImageDataset.create(
+            display_name=f"{self.action.config_data.get('model_name')}",
+            gcs_source=vertexai_import_file,
+            import_schema_uri=aiplatform.schema.dataset.ioformat.image.bounding_box,
+        )
+
+        response = self.train_automl_model(credentials, working_dataset)
+
+        model_name = response.__dict__['_pb'].name
+        id_separator_index = model_name.rindex('/')
+        model_id = model_name[id_separator_index + 1:]
+        
+        ActionRun.set_action_run_status(self.session, self.action_run.id, "finished")
+
+        return {
+            "success": True,
+            "model_name": self.action.config_data.get('model_name'),
+            "model_id": model_id
+        }
+
+
+    def get_file_list(self, session):
+        directory_id = self.action.config_data.get('directory_id')
+        self.directory = WorkingDir.get_by_id(session = session, directory_id=directory_id)
+        file_list = WorkingDirFileLink.file_list(session=session, working_dir_id=self.directory.id, limit=None)
+
+        return file_list
+
+
+    def build_vertex_format_instance(self, instance, session, file):
+        label = File.get_by_id(session=session, 
+                               file_id=instance.label_file_id)
+
+        image_height = file.image.height
+        image_width = file.image.width
+        if label is not None and self.label_count.get(label.label.name) > 10:
+            vertex_format_instance = {
+                "displayName": label.label.name,
+                "xMin": instance.x_min/image_width,
+                "xMax": instance.x_max/image_width,
+                "yMin": instance.y_min/image_height,
+                "yMax": instance.y_max/image_height
+            }
+            return vertex_format_instance
+
+    def build_vertexai_instance_file_instance_array(self, file, session):
```

Comment:
I really like that you are building more concise functions instead of a really big function! This helps a lot with code readability :D